clouddrift/binning.py

```diff
@@ -0,0 +1,177 @@
+"""Module for binning Lagrangian data."""
+
+from collections.abc import Iterable
+
+import numpy as np
+import xarray as xr
+
+DEFAULT_BINS_NUMBER = 10
+
+
+def binned_statistics(
+    coords: np.ndarray | list[np.ndarray],
+    data: np.ndarray | list[np.ndarray] | None = None,
+    bins: int | list = DEFAULT_BINS_NUMBER,
+    bins_range: list | None = None,
+    dim_names: list[str] | None = None,
+    output_names: list[str] | None = None,
+    zeros_to_nan: bool = False,
+) -> xr.Dataset:
+    """
+    Compute N-dimensional histogram binning and calculate variable means in each bin.
+
+    Parameters
+    ----------
+    coords : array-like or list of array-like
+        Arrays of the coordinates of the Lagrangian data to be binned. For 1D data, a single array can be provided.
```

Comment:
Array(s) of Lagrangian data coordinates to be binned. For 1D, provide a single array. For N-dimensions, provide a list of N arrays, each giving coordinates along one dimension.